### PR TITLE
bump up place admin_levels to make room for region

### PR DESCRIPTION
**Pull request by pleary**

### CODE REVIEW

File: app/assets/javascripts/atlases/show.js.erb
Comment (kueda) [inline app/assets/javascripts/atlases/show.js.erb:166]: Since this is an ERB can we just refer to `Place::COUNTY_LEVEL`? Or are we trying to avoid model references in the JS templates?

File: app/models/observation.rb
Comment (kueda) [inline app/models/observation.rb:2663]: I think the problem here was either a circular reference problem, or maybe loading the Place model triggered a lot of other stuff like loading preferences that then caused problems. Could we avoid that by moving these constance into a standalone module or class?

File: app/webpack/observations/identify/ducks/suggestions.js
Comment (kueda) [inline app/webpack/observations/identify/ducks/suggestions.js:125]: Would be nice to find a better place for these constants in JS, though clearly I didn't do that when I wrote this stuff originally. Maybe a `.js.erb` that gets included in the application bundle? Something to consider in the future, maybe.

File: db/migrate/20210930182050_modify_place_admin_levels.rb
Comment (kueda) [inline db/migrate/20210930182050_modify_place_admin_levels.rb:45]: This looks pretty chunky. Any estimate on how long it will take?

**Review by kueda — APPROVED** [review]
Looks good to me. Most of my comments are about trying to keep the constants in one place, but I don't think that needs to block merging.

File: app/assets/javascripts/atlases/show.js.erb
Comment (pleary) [reply app/assets/javascripts/atlases/show.js.erb:166]: Yeah I think we probably could use the model constant here. I'll try that out

File: app/models/observation.rb
Comment (pleary) [reply app/models/observation.rb:2663]: I didn't try any other options here because of the comment, and I wasn't sure exactly what issues it was causing. I can look into it more, but maybe after merging and deploying in this code so we can isolate this specific change

File: app/webpack/observations/identify/ducks/suggestions.js
Comment (pleary) [reply app/webpack/observations/identify/ducks/suggestions.js:125]: We often define constants like SITE, CURRENT_USER, LIFE_TAXON, etc in views when rendering React components. So maybe we could extract those as well as other things we use more than once like these admin_level contstants into a JS erb file

File: db/migrate/20210930182050_modify_place_admin_levels.rb
Comment (pleary) [reply db/migrate/20210930182050_modify_place_admin_levels.rb:45]: The place one is pretty fast. Users is a bit longer, which is why I'm using a smaller batch size of 1000 so the update queries don't take too long. I don't have an exact time - I'm hoping to get that when running on staging. But I suspect it'll be on the order of a few minutes - less than 15 minutes total

Comment (kueda) [reply db/migrate/20210930182050_modify_place_admin_levels.rb:45]: Ah, so you're expecting to do this with some downtime?

Comment (pleary) [reply db/migrate/20210930182050_modify_place_admin_levels.rb:45]: I wasn't planning on doing this with downtime, but we can if you think that'll be preferred. There will be a short period of time when the admin_levels in the DB are updated, but the code the apps are using don't reflect the latest admin_levels since we'll migrate before restarting the apps. That may result in a few small issues for things like the dashboards of users w/ no activity, but I figured that downtime would be more disruptive. Now that I think of it, it's probably best to update the users admin_level first since that'll take a little longer and that change is least likely to result in problems, so I'll change the order in this migration

Comment (kueda) [reply db/migrate/20210930182050_modify_place_admin_levels.rb:45]: Hm, well I guess if things don't get locked up and Capistrano isn't going to hang during a 15 min deploy then this sounds ok to me.

**Pull request merged**
